### PR TITLE
Adds explosive bracelets

### PR DESCRIPTION
**Pull request by GuppyLaxx**
What it says on the title.
It's essentially a "ghetto freedom implant", takes up your gloves slot which means no insulateds, also can't be taken off.
**Takes 5 seconds to activate when put on, which locks it to the arm. After that you can activate it at any point like any other implant, when activated it beeps 3 times(takes 4.5 seconds) then blows your arm off, knocking the person down, hurting them a bit, and flashing anyone nearby.**
Recipe can be changed, could also make it so you can "disable it" and take it off, etc
Current recipe : 1 IED, 1 igniter, 5 cable, 2 metal, 1 micro laser. Requires a screwdriver.
Sprites from Navi.OS on discord
image
:cl: GuppyLaxx and NaviOS
add: added explosive bracelets. simple as that.
/:cl:
ps : i can clean up the code more to avoid certain snowflake cases if anyone wants, but this works, so

### CODE REVIEW

File: hippiestation/code/modules/clothing/gloves/explosivegloves.dm [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	icon = 'hippiestation/icons/obj/clothing/gloves.dmi'
+	alternate_worn_icon = 'hippiestation/icons/mob/hands.dmi'
+	actions_types = list(/datum/action/item_action/hands_free/exbracelet)
+	icon_state = "exbracelet_unequipped"
+	item_state = "exbracelet_right"
+	var/armed = FALSE
+	var/wrist = "right"
+
+/obj/item/clothing/gloves/exbracelet/Initialize()
+	. = ..() //i still dont know what the . = is for lmao
```
Comment (yoyobatty) [inline hippiestation/code/modules/clothing/gloves/explosivegloves.dm]: . is the default return value. The byond 'help-on' reference in dream maker describes this.

File: hippiestation/code/modules/clothing/gloves/explosivegloves.dm [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	sleep(15)
+	to_chat(user, "<span class='bold'>Kiss your [wrist] arm a goodbye and say hello to freedom!</span>")
+	var/obj/item/bodypart/arm
+	if(wrist == "left")
+		arm = user.get_bodypart("l_arm")
+	else
+		arm = user.get_bodypart("r_arm")
+	arm.dismember()
+	user.apply_damage(10, BRUTE, "chest") //because the explosion or whatever
+	user.Knockdown(10)
```
Comment (yoyobatty) [inline hippiestation/code/modules/clothing/gloves/explosivegloves.dm:69]: that's a 1 second knockdown

**Comment by yoyobatty** [comment]
Have you attempted to "break" it aka spam the activate button or activate it and then try to remove it?

File: hippiestation/code/modules/clothing/gloves/explosivegloves.dm [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+/obj/item/clothing/gloves/exbracelet/equipped(mob/user, slot)
+	..()
+	if(slot == SLOT_GLOVES)
+		icon_state = "exbracelet_unarmed"
+		to_chat(user, "<span class='warning'>5 seconds until [src] is armed!</span>")
+		addtimer(CALLBACK(src, .proc/activate, user), 50)
+
+/obj/item/clothing/gloves/exbracelet/proc/activate(mob/user)
+	item_flags = NODROP
```
Comment (yoyobatty) [inline hippiestation/code/modules/clothing/gloves/explosivegloves.dm]: it's probably better to use item_flags |= NODROP

File: hippiestation/code/modules/clothing/gloves/explosivegloves.dm (hunk 2 above)
Comment (GuppyLaxx) [reply hippiestation/code/modules/clothing/gloves/explosivegloves.dm:69]: yes, i know

**Comment by GuppyLaxx** [comment]
if you wanna remove it after activating it by cutting your arm off to shit on the code, be my guest.
spamming it's just gonna runtime and kill the user if they spammed it enough

**Comment by JohnGinnane** [comment]
lovely

**Comment by JohnGinnane** [comment]
It doesn't look like you check if someone takes the bracelet off within 5 seconds of putting it on?

**Comment by Carbonhell** [comment]
The idea itself seems cool, but this pretty much gives greytiders the tools to be even more problematic, even though it's a shitty freedom implant I still have doubts on the idea.

**Comment by JohnGinnane** [comment]
I think it's high risk high reward, they lose an arm to do this, and it stuns everyone equally so competent players will just be able to stun them again

**Comment by GuppyLaxx** [comment]
Actually, it only stuns the user. For a second, the flash is just the white screen and `very` small eye damage.
Agaisnt an officer? They have time to react by pulling their baton and getting them while they're blind, even while down for that one second. Sunglasses protect.
Agaisnt an antag? Now, MOST don't get sunglasses by default, yet they can still get them, this gives random greytiders some leeway to escape a really shitty antag cuffing and then going `lol now ur dead kiddo haha u have no headset watch as you slowly suffer`

**Comment by GuppyLaxx** [comment]
also i added the micro laser to the recipe so greytiders needed a lathe atleast to get one rather than mass produce straight from the tool room

**Comment by JohnGinnane** [comment]
Oh sorry, I thought you stunned and flashed people in range, my :bee:

**Comment by Manly-Dwarf** [comment]
I kind of like it, but because it works as a freedom implant and also blows off your arm, it means you effectively can't be cuffed at all any more. If it just dealt damage it'd be better. Especially because it can't be taken off of you. Maybe if we added an implant disabler to sec or something it'd be better, but that just adds even more balance issues onto that heap.

**Comment by GuppyLaxx** [comment]
it deals 10 brute damage to the chest meaning if you're already hurt enough before you'll be slowed down, and knocks you down for 1 second, 1 second is more than enough for a sec officer to baton you, pull out the baton, taze you, whatever.
this thing justifies the existence of cuff shoes/straight jackets more atleast

**Comment by Carbonhell** [comment]
I'm sorry, but I don't really like this idea and I cannot find a way to not make it powergame-centered. I have to put a -1 on this.

**Comment by ghost** [comment]
i gotta say -1
since it removes the arm it'll prove a huge annoyance and have to deal with even more shit if they wanna deal with greytide

**Comment by Carbonhell** [comment]
I am very sorry man
Denied

**Pull request closed without merging**